File: Behavioral/Chain of responsibility/Delphi/FirstHandler.pas
unit FirstHandler;

interface

uses
  BaseHandler;

type
  TFirstHandler = class(TBaseHandler)
    procedure Handle(request: string); override;
  end;

implementation

procedure TFirstHandler.Handle(request: string);
begin
  if Length(request) > 5 then
    Writeln('The first processor processed:', request)
  else
    inherited Handle(request);
end;

end.

File: Behavioral/Chain of responsibility/C#/ChainOfResponsibilityExample/ChainOfResponsibilityExample/FirstHandler.cs
namespace ChainOfResponsibilityExample;

class FirstHandler : BaseHandler
{
	public override void Handle(string request)
	{
		if (request.Length > 5)
		{
			Console.WriteLine("The first processor processed: " + request);
		}
		else
		{
			base.Handle(request);
		}
	}
}